File: KIJUNG/PRO_WEEK_02/숫자문자열과영단어.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string s) {
    int answer = 0;
    string nums[10] = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};

    for(int i=0;i<s.size();i++){
        if(s[i] <= '9' && s[i] >=0){
            answer = answer*10+(s[i]-48);
        }
        else{
            for(int j=0;j<10;j++){
                if(i+num[j].size()<=s.size() && num[j] == s.substr(i,num[j].size)) {
                    answer = answer*10+j;
                    i+=num[j].size()-1;
                    break;
                }
            }
        }
    }
    return answer;
}

File: DONGUK/프로그래머스_WEEK 2/숫자 문자열과 영단어.py
def solution(s):
    num_dict = {'zero': '0', 'one':'1', 'two':'2', 'three':'3', 'four':'4', 
                'five': '5', 'six':'6', 'seven':'7','eight':'8', 'nine':'9'}
    
    nums = ['0','1','2','3','4','5','6','7','8','9']
    
    answer = ''
    
    s = list(s)
    
    tmp = ''
    for ch in s:
        if ch not in nums:
            tmp += ch
        
            if tmp in num_dict:
                answer += num_dict[tmp]
                tmp = ''
        else:
            answer += ch
    
    answer = int(answer)
    return answer